blacktie/jatmibroker-xatmi/src/main/java/org/jboss/narayana/blacktie/jatmibroker/core/transport/hybrid/stomp/StompManagement.java

```diff
@@ -81,14 +81,17 @@ public static void send(Message message, OutputStream outputStream) throws IOExc
             toSend.append("\n");
         }
         toSend.append("\n");
-        outputStream.write(toSend.toString().getBytes());
+        synchronized (outputStream) {
+            outputStream.write(toSend.toString().getBytes());
```

Comment:
This was the comment I made on the previous outdated diff:

I also think it is best to assemble the data into the buffer and then write it to the socket in one go. So using getChannel and assembling the data before writing we would have:

    ByteBuffer toSend = ByteBuffer.wrap(message.getCommand().getBytes());

    if (message.getBody() != null)
        toSend.put(message.getBody());

    toSend.put("\000\n\n".getBytes()); 

    socket.getChannel().write(toSend);
Implementing in this way avoids the need for synchronization.